python/tests/test_tables.py

```diff
@@ -2272,19 +2272,231 @@ class TestSubsetTables(unittest.TestCase):
 
     seed = 12345
 
+    def subset_ragged_columns(self, packed, offset, subset, name):
+        """
+        Subsets a pair of ragged columns by the rows specified in `subset`.
+        :param numpy.ndarray packed: The flattened array of data.
+        :param numpy.ndarray offset: The array of offsets into the ``packed`` array.
+        : param str name: The name of the ragged columns.
+        :return: A tuple with the subsetted flattened array of data and offsets.
+        :rtype: tuple[numpy.ndarray]
+
+        """
+        if name in ["ancestral_state", "derived_state", "record", "timestamp"]:
+            unpack = tskit.unpack_strings
+            pack = tskit.pack_strings
+        elif name == "location":
+            unpack = tskit.unpack_arrays
+            pack = tskit.pack_arrays
+        else:
+            unpack = tskit.unpack_bytes
+            pack = tskit.pack_bytes
+        unpacked = unpack(packed, offset)
+        unpacked = [unpacked[i] for i in subset]
+        return pack(unpacked)
+
+    def np_subset(self, tables, nodes, record_provenance=True):
+        """
```

Comment:
We don't want to duplicate documentation. Just leave this at "Numpy version of TableCollection.subset, for testing purposes."